src/mlpack/core/util/io.hpp

```diff
@@ -322,9 +390,20 @@ class IO
   //! So that Timer::Start() and Timer::Stop() can access the timer variable.
   friend class Timer;
 
-  //! Pointer to the ProgramDoc object.
-  util::ProgramDoc* doc;
+  //! Pointer to the ProgramName object.
+  util::ProgramName* pname;
+
+  //! Pointer to the ShortDescription object.
+  util::ShortDescription* shortDesc;
+
+  //! Pointer to the LongDescription object.
+  util::LongDescription* longDesc;
+
+  //! Pointer to the Example object.
+  std::vector<util::Example*> examples;
 
+  //! Pointer to the SeeAlso object.
+  std::vector<util::SeeAlso*> seeAlsos;
```

Comment:
Let me know what you think, but I think maybe it might be easier to maintain this with some kind of `BindingDetails` struct or something that collects all of this information in it.  It would also reduce the number of methods added to `IO`.  However I am not sure if there's a technical difficulty that I've missed in thinking about this---but I think it *should* be possible to wrap all these in one class without any problem. :+1:

Reply:
I have tried this approach in some last commit, Let know what do you think?

Reply:
Looks great! :+1: